.cargo/config

```diff
@@ -1,5 +1,6 @@
 [alias]
 xtask = "run --package xtask --"
+ruma-clippy = "clippy -p ruma --all-targets --all-features --quiet -- -W clippy::cloned_instead_of_copied -W clippy::dbg_macro -W clippy::empty_line_after_outer_attr -W clippy::inefficient_to_string -W clippy::macro_use_imports -W clippy::map_flatten -W clippy::mut_mut -W clippy::needless_borrow -W clippy::print_stdout -W clippy::str_to_string -W clippy::todo -W clippy::unreadable_literal -W clippy::unseparated_literal_suffix -W clippy::useless_let_if_seq -W clippy::wildcard_imports"
```

Comment:
You can use a multiline string

```suggestion
ruma-clippy = """\
  clippy -p ruma --all-targets --all-features --quiet -- \
    -W clippy::cloned_instead_of_copied \
    -W clippy::dbg_macro \
    -W clippy::empty_line_after_outer_attr \
    -W clippy::inefficient_to_string \
    -W clippy::macro_use_imports \
    -W clippy::map_flatten \
    -W clippy::mut_mut \
    -W clippy::needless_borrow \
    -W clippy::print_stdout \
    -W clippy::str_to_string \
    -W clippy::todo \
    -W clippy::unreadable_literal \
    -W clippy::unseparated_literal_suffix \
    -W clippy::useless_let_if_seq \
    -W clippy::wildcard_imports \
  """
```

Reply:
Also I think non-clippy lints can be enabled, could you add `-W rust-2018-idioms` before the `-W clippy::`s? Could also go through https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html to look for others that might be worth enabling.

Reply:
Another one (that I'm now removing from crate attrs because I expect we'll add it here): `missing-debug-implementations`.

Reply:
One more thing, we should have `-A clippy::new-without-default` in there.

Reply:
After that and having `.clippy.toml` with an msrv, we should be able to remove every `#![allow(clippy::foobar)]`.

Reply:
> Another one (that I'm now removing from crate attrs because I expect we'll add it here): missing-debug-implementations.

This fails because of test code and `ruma_serde::urlencode` do we still want this?

Reply:
Let's leave it out for now.